JavaScript/chapter02/p01_remove_dups/avc278.js

```diff
@@ -0,0 +1,110 @@
+// Remove Dups: Write code to remove duplicates from an unsorted linked list.
+// Follow Up: How would you solve this problem if a temporary buffer is not allowed?
+
+const assert = require("assert");
+
+class LinkedListNode {
+  constructor(val, next) {
+    this.val = val === undefined ? null : val;
+    this.next = next === undefined ? null : next;
+  }
+}
+
+const arrayToLinkedList = (arr) => {
+  let tail = null;
+  for (let i = arr.length - 1; i >= 0; i--) {
+    tail = new LinkedListNode(arr[i], tail);
+  }
+  return tail;
+};
+
+const compareLinkedLists = (A, B) => {
```

Comment:
what should the following expression evaluate to?

```javascript
compareLinkedLists(arrayToLinkedList([]), arrayToLinkedList(['hi!']))
```

Reply:
Oh interesting. That didn't even occur to me to try initially. Good catch.
This will definitely be fixed in the next commit.